lib/improver/nbhood/nbhood.py

```diff
@@ -203,7 +203,7 @@ def process(self, cube, mask_cube=None):
         # the cube is sliced, so that the realization becomes a scalar
         # coordinate.
         try:
-            realiz_coord = cube.coord('realization')
+            realiz_coord = cube.coord('realization', dim_coords=True)
```

Comment:
Additionally checks that realization is a dim-coord.
@benfitzpatrick: Should this change be reflected in the comment above?

Reply:
Basically it is a double check that it is not scalar

Reply:
Updated comment